pkg/kubelet/kubeclientmanager/kubeclient_manager.go

```diff
@@ -77,28 +82,41 @@ func (manager *KubeClientManager) RegisterTenantSourceServer(source string, ref
 		return
 	}
 
-	if _, ok := manager.tenant2api[key]; !ok {
-		manager.tenant2api[key] = clientId
-		klog.Infof("added %d to the map, map has %+v", clientId, manager.tenant2api)
+	if _, ok := manager.puid2api[ref.UID]; !ok {
+		manager.puid2api[ref.UID] = clientId
+		klog.V(6).Infof("added %d to the map, map has %+v", clientId, manager.puid2api)
 	}
 }
 
-func (manager *KubeClientManager) GetTPClient(kubeClients []clientset.Interface, tenant string) clientset.Interface {
+func (manager *KubeClientManager) UnregisterTenantSourceServer(ref *v1.Pod) {
+	if len(ref.UID) == 0 {
+		klog.Warningf("unable to unregister tenant source for pod with empty UID: tenant='%s'", ref.Tenant)
```

Comment:
unable to unregister api source for pod?

Reply:
will rephrase